src/StaticAnalyzers.cmake

```diff
@@ -54,14 +64,6 @@ function(_enable_clang_tidy_setup_cl CXX_FLAGS C_FLAGS)
   set(CLANG_TIDY_CXX_FLAGS ${CXX_FLAGS})
   set(CLANG_TIDY_C_FLAGS ${C_FLAGS})
 
-  if(CMAKE_CXX_STANDARD)
```

Comment:
Why clang tidy args are removed?

Reply:
I have to remove it, otherwise clang-tidy wrongly sets`-std=c++23` according to `CMAKE_CXX_STANDARD` while clang 15 only supports `-std=c++2b` ([failure example](https://github.com/aminya/project_options/actions/runs/10371430640/job/28711716651#step:5:1380)). This change is ok, since clang-tidy knows the actual compile flags (including `-std=<version>`) according to [the cmake implementation code](https://gitlab.kitware.com/cmake/cmake/-/blob/8733dcd11588f2d9eb0bddc8af8a6db7a2e9eba8/Source/cmcmd.cxx#L370-416), especially [this comment](https://gitlab.kitware.com/cmake/cmake/-/blob/8733dcd11588f2d9eb0bddc8af8a6db7a2e9eba8/Source/cmcmd.cxx#L382-389).